docusaurus/docs/Android/03-ui/04-message-components/04-message-input.mdx

```diff
@@ -45,23 +45,30 @@ viewModel.bindView(messageInputView, viewLifecycleOwner)
 
 ## Handling Actions 
 
-Actions can be handled by setting listeners on this view. For example, you can handle clicks of the send button or when the user starts and stops typing:
+Actions can be handled by setting listeners on this view. For example, you can handle clicks of the send button:
 
 ```kotlin
 messageInputView.setOnSendButtonClickListener {
     // Handle send button click
 }
+```
 
-messageInputView.setTypingListener(
-    object : MessageInputView.TypingListener {
-        override fun onKeystroke() {
-            // Handle keystroke case
-        }
+If you want to listen to typing updates, you can set your own  [`DefaultTypingUpdatesBuffer`](https://github.com/GetStream/stream-chat-android/blob/main/stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/utils/typing/DefaultTypingUpdatesBuffer.kt) on the View.
+The buffer is designed to save valuable API calls by not firing events unless they are distinct. Additionally, it automatically tracks typing inactivity after the last keystroke and sends a stop typing event after the time defined by `delayIntervalMillis` has elapsed.
 
-        override fun onStopTyping() {
-            // Handle stop typing case
+```kotlin
```

Comment:
Unsure where mentioning the Compose counterpart of this would fit in docs, as we dedicate the docs to the UI and not to the VMs and controllers which is where this resides in Compose.

Reply:
We could fit it into the MessageList or MessageCompose docs, and touch upon using the ViewModel and/or its features (in this case the typing).

Reply:
Moved to the composer docs